internal/k8s/service/resolver.go

```diff
@@ -290,26 +297,54 @@ func (r *backendResolver) findGlobalCatalogService(service *corev1.Service) (*Re
 		r.logger.Trace("error retrieving nodes", "error", err)
 		return nil, err
 	}
-	filter := fmt.Sprintf(`Meta[%q] == %q and Meta[%q] == %q and Kind != "connect-proxy"`, MetaKeyKubeServiceName, service.Name, MetaKeyKubeNS, service.Namespace)
-	for _, node := range nodes {
-		nodeWithServices, _, err := r.consul.Catalog().Node(node.Node, &api.QueryOptions{
-			Filter: filter,
-		})
-		if err != nil {
-			r.logger.Trace("error retrieving node services", "error", err, "node", node.Node)
+
+	var consulNamespaces []*api.Namespace
+	// the default namespace
+	namespaces := []string{""}
+	consulNamespaces, _, err = r.consul.Namespaces().List(nil)
+	if err != nil {
+		if !strings.Contains(err.Error(), "Unexpected response code: 404") {
+			r.logger.Trace("error retrieving namespaces", "error", err)
```

Comment:
Could this return other errors like a 5xx if ACLs are wrong? (The string matching instead of checking a status code feels 😬 but IIRC that's an upstream issue with `consul/api`)

Reply:
in which case that's completely fine, the 404 is mainly due to the fact that OSS consul will give a 404 for the entirety of that endpoint -- it's what I mentioned as a dirty hack to tell whether you're using OSS or enterprise -- basically if we don't have that API endpoint we should still loop once over the empty "namespace" since we're on OSS

Reply:
Ooh, okay - that could probably use a comment explaining what's going on there.